internal/services/nginx/nginx_configuration_resource.go

```diff
@@ -36,10 +36,10 @@ type ProtectedFile struct {
 	VirtualPath string `tfschema:"virtual_path"`
 }
 
-func (c ProtectedFile) toSDKModel() nginxconfiguration.NginxConfigurationFile {
-	return nginxconfiguration.NginxConfigurationFile{
-		Content:     pointer.To(c.Content),
-		VirtualPath: pointer.To(c.VirtualPath),
+func (c ProtectedFile) toSDKModel() nginxconfiguration.NginxConfigurationProtectedFileRequest {
+	return nginxconfiguration.NginxConfigurationProtectedFileRequest{
+		Content:     pointer.FromString(c.Content),
+		VirtualPath: pointer.FromString(c.VirtualPath),
```

Comment:
This should remain as 
```suggestion
		Content:     pointer.To(c.Content),
		VirtualPath: pointer.To(c.VirtualPath),
```

Also shouldn't `content` be deprecated in the resource as well?

Reply:
We still want the user to be able to provide the content of the file. It's the output that they get back will no longer be the case.